packages/flutter/lib/src/widgets/navigator.dart

```diff
@@ -2408,6 +2491,21 @@ class _RouteEntry extends RouteTransitionRecord {
   Route<dynamic> lastAnnouncedPoppedNextRoute = notAnnounced; // last argument to Route.didPopNext
   Route<dynamic> lastAnnouncedNextRoute = notAnnounced; // last argument to Route.didChangeNext
 
+  /// Restoration ID used for the encapsulated route if restoration is enabled
+  /// for it or null if restoration cannot be enabled for this route.
+  String get restorationId {
+    // User-provided restoration ids of Pages are prefixed with 'p'. Generated
+    // ids for pageless routes are prefixed with r to avoid clashes.
```

Comment:
put some punctuation between the prefix and the id (or use something other than letters), to avoid the risk of someone being offended if they use the id "rick" or something.

Reply:
lol, but good point. Added a "+".